hadoop-hdds/server-scm/src/main/resources/webapps/scm/scm.js

```diff
@@ -82,12 +82,16 @@
                 });
             /*if option is 'All' display all records else display specified record on page*/
             $scope.UpdateRecordsToShow = () => {
-                if($scope.RecordsToDisplay == 'All') {
-                    $scope.lastIndex = 1;
-                    $scope.nodeStatus = nodeStatusCopy;
-                } else {
-                    $scope.lastIndex = Math.ceil(nodeStatusCopy.length / $scope.RecordsToDisplay);
-                    $scope.nodeStatus = nodeStatusCopy.slice(0, $scope.RecordsToDisplay);
+                if($scope.search){
+                    $scope.nodeStatus = nodeStatusCopy.filter(item => item.hostname.includes($scope.search));
+                }else {
+                    if ($scope.RecordsToDisplay == 'All') {
```

Comment:
It's a good practice to use strict equality (===) instead of loose equality (==) in JavaScript to avoid unexpected type coercion.

Reply:
Thanks for your suggestion @ArafatKhan2198 , I am already  refactor to the function, with a few changed comments.